internal/goofys.go

```diff
@@ -702,7 +702,18 @@ func (fs *Goofys) ForgetInode(
 		fs.mu.Unlock()
 
 		if inode.Parent != nil {
-			inode.Parent.removeChild(inode)
+			inode.Parent.mu.Lock()
+			fs.mu.Lock()
+			// get inode and its parent again in case other goroutines removed them before we get the locks
+			inode := fs.inodes[op.Inode]
+			if inode != nil && inode.Parent != nil {
+				parent := fs.inodes[inode.Parent.Id]
+				if parent != nil {
+					parent.removeChildUnlocked(inode)
```

Comment:
As shown in the stack trace, there is at least one race condition where we try to remove inode from the parent but it's already removed.